### ISSUE TITLE
Authorization intermittently fails due to invalid sessionKey

### ISSUE THREAD

**Issue by harunsmrkovic** [digime/digime-sdk-ios#132]
I intermittently get a failed auth from the digi.me app

```
2019-10-11 14:33:48.588 [info][tid:com.facebook.react.JavaScript] [digi.me] Initiated authorization via SDK
2019-10-11 14:33:48.589 [info][tid:com.facebook.react.NativeBridgeQueue][NativeBridge.m:46] [digime] Authorize inited
[Instabug] Attempting to identify user with an invalid email. User will not be identified.
2019-10-11 14:34:05.838 [warn][tid:main][NativeBridge.m:52] [digime] Authorization failed digi.me app returned an invalid session key.
```

**Comment by harunsmrkovic** [comment]
I got this again, but getAccounts call seems to work, even though SDK reports auth failure.

```
2019-10-14 19:59:03.740 [warn][tid:main][NativeBridge.m:52] [digime] Authorization failed digi.me app returned an invalid session key.
2019-10-14 19:59:04.174 [info][tid:com.facebook.react.JavaScript] [digime] Initiated getAccounts via SDK
2019-10-14 19:59:04.176 [info][tid:com.facebook.react.NativeBridgeQueue][NativeBridge.m:70] [digime] Get Accounts inited
2019-10-14 19:59:04.434 [info][tid:com.facebook.react.JavaScript] '[digime] Got Accounts', '[{"id":"1_114286873000913","name":"Joseph Clubb","service":{"name":"Facebook","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/production\\/discovery\\/services\\/facebook\\/icon75x75.png"}},{"id":"4_10874014787","name":"joseph.clubb","service":{"name":"Instagram","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/production\\/discovery\\/services\\/instagram\\/icon75x75.png"}},{"id":"18_7GYFKW","name":"Joseph C.","service":{"name":"Fitbit","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/production\\/discovery\\/services\\/fitbit\\/icon75x75.png"}},{"id":"32_k6Lr4myx4XHn1eDmo6yAfGjdGwqMwBhWlkdJb","service":{"name":"Plaid","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/development\\/discovery\\/providers\\/usabanks\\/icon75x75.png"}},{"id":"32_j91KagbJa3ToAbrMn9XmIEQ1Wj7Vr1u1mkG3Z","number":"1111","name":"Plaid Silver Standard 0.1% Interest Saving","service":{"name":"Chase","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/development\\/discovery\\/providers\\/usabanks\\/icon75x75.png"}},{"id":"32_eG78zJvPz4upjo63DGmqI3zeXlp4gefLBKVwE","number":"3333","name":"Plaid Diamond 12.5% APR Interest Credit Card","service":{"name":"Chase","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/development\\/discovery\\/providers\\/usabanks\\/icon75x75.png"}},{"id":"19_0j648agi0cuu0a5qzlribml82","name":"digi.me office","service":{"name":"Spotify","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/production\\/discovery\\/services\\/spotify\\/icon75x75.png"}},{"id":"31_UCUIv_KFaSvY2FuGJMQmp3JQ","name":"digi.me","service":{"name":"YouTube","logo":"https:\\/\\/securedownloads.digi.me\\/static\\/production\\/discovery\\/services\\/youtube\\/icon75x75.png"}}]'```

`digi.me app returned an invalid session key` is what is returned from `authorizeWithCompletion -> error.localizedDescription`

**Comment by brokenzz** [comment]
This stems from having an out of date auth request coming back to the SDK. This could occur if one of the auth requests was left unhandled by the user (neither approved nor denied). As a result, the SDK would hold on to the old auth handler, which would be used first to validate the latest auth request, creating a mismatch.

Added a fix in #138 which ensures that there's only ever 1 active auth response handler. This will be included in the next SDK version (3.0.1). You don't need to do anything other than update to it when it is available.

Secondary to this, we discovered a bug in the digi.me app where subsequent auth request would not clear previous "undecided" contract from view. This will be fixed in the next version of digi.me app.
